server/graphql/v2/query/collection/OrdersCollectionQuery.ts

```diff
@@ -235,7 +249,27 @@ export const OrdersCollectionResolver = async (args, req: express.Request) => {
 
   const fetchAccountParams = { loaders: req.loaders, throwIfMissing: true };
   const host = args.host && (await fetchAccountWithReference(args.host, fetchAccountParams));
-  let account, oppositeAccount, hostedAccounts;
+  let account, oppositeAccount, hostedAccounts, includeHostedAccounts, excludeHostAccount;
+
+  switch (args.hostContext) {
+    case 'ALL':
+      includeHostedAccounts = true;
+      excludeHostAccount = false;
+      break;
+    case 'HOSTED':
+      includeHostedAccounts = true;
+      excludeHostAccount = true;
+      break;
+    case 'INTERNAL':
+    default:
+      includeHostedAccounts = false;
+      excludeHostAccount = false;
+  }
+  // Override with deprecated includeHostedAccounts argument
+  if (args.includeHostedAccounts) {
+    includeHostedAccounts = args.includeHostedAccounts;
+  }
```

Comment:
To handle the case where we'd be passing `false`

```suggestion
  if (!isNil(args.includeHostedAccounts)) {
    includeHostedAccounts = args.includeHostedAccounts;
  }
```